### PR TITLE
Support specifying an execution provider in benchmark script

### PR DESCRIPTION
**Pull request by zhangyaobit**
**Description**: Describe your changes.
Support specifying an execution provider in benchmark script.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/python/tools/transformers/bert_perf_test.py [hunk 1]
```diff
@@ -61,8 +62,19 @@ def create_session(model_path, use_gpu, intra_op_num_threads, graph_optimization
...
     else:
-        execution_providers = ['CPUExecutionProvider'
-                               ] if not use_gpu else ['CUDAExecutionProvider', 'CPUExecutionProvider']
+        if use_gpu:
+            if provider == 'dml':
+                execution_providers = ['DmlExecutionProvider', 'CPUExecutionProvider']
+            elif provider == 'rocm':
+                execution_providers = ['ROCMExecutionProvider', 'CPUExecutionProvider']
+            elif provider == 'migraphx':
+                execution_providers = ['MIGraphXExecutionProvider', 'CPUExecutionProvider']
```
Comment (weixingzhang) [inline onnxruntime/python/tools/transformers/bert_perf_test.py:71]: Falling back to ROCm EP and then CPU EP: ['MIGraphXExecutionProvider', 'ROCMExecutionProvider', 'CPUExecutionProvider']

File: onnxruntime/python/tools/transformers/bert_perf_test.py [hunk 2]
```diff
@@ -61,8 +62,19 @@ def create_session(model_path, use_gpu, intra_op_num_threads, graph_optimization
...
+            elif provider == 'tensorrt':
+                execution_providers = ['TensorrtExecutionProvider', 'CPUExecutionProvider']
```
Comment (weixingzhang) [inline onnxruntime/python/tools/transformers/bert_perf_test.py:73]: Similar here. Falling back to CUDA EP and then CPU EP.

File: onnxruntime/python/tools/transformers/bert_perf_test.py (hunk 1 above)
Comment (zhangyaobit) [reply onnxruntime/python/tools/transformers/bert_perf_test.py]: Done.

File: onnxruntime/python/tools/transformers/bert_perf_test.py (hunk 2 above)
Comment (zhangyaobit) [reply onnxruntime/python/tools/transformers/bert_perf_test.py]: Done.

**Review by weixingzhang — APPROVED** [review]

**Review by weixingzhang — APPROVED** [review]

**Pull request merged**